### DISCUSSION TITLE
Dielectric material functions using uniaxial anisotropic materials

### DISCUSSION THREAD

**Discussion by chuksmbaka** [NanoComp/meep#2335] [Q&A]
### Discussed in https://github.com/NanoComp/meep/discussions/1894

Halo,
 I will like to define a material function for the dielectric constant for a uniaxial anisotropic material. I have issues with providing a vector argument with the function and assigning the various dielectric constants to this position. The code is as below

```
import meep as mp

resolution = 10
pml_layer = [mp.PML(thickness=0.2)]
cell = mp.Vector3(10, 10, 10)


def LiN(p):
    eps_o = 100
    eps_e = 100
    ln_epsilon = mp.Matrix(mp.Vector3(eps_o, 0, 0), mp.Vector3(0, eps_o, 0), mp.Vector3(0, 0, eps_e))
    ln_epsilon_diag = mp.Vector3(ln_epsilon[0].x, ln_epsilon[1].y, ln_epsilon[2].z)
    return mp.Medium(epsilon_diag=ln_epsilon_diag)

```

**Comment by smartalecH** [comment]
Try this:

```
def LiN(p):
    # put some logic here about the shape of your object...
    # for example: if p.x == 1:
    eps_o = 100
    eps_e = 100
    return mp.Medium(epsilon_diag=mp.Vector3(eps_o, eps_o, eps_e))
```

As discussed in the [docs](https://meep.readthedocs.io/en/latest/Python_User_Interface/#medium), `epsilon_diag` is a vector corresponding to the *diagonal* of the material tensor... so you don't need to go through the hassle of building a matrix and extracting the diagonal...
